sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/application-offering-change-request.aest.controller.ts

```diff
@@ -105,14 +111,42 @@ export class ApplicationOfferingChangeRequestAESTController extends BaseControll
    * Approves or declines an application offering change request.
    * @param applicationOfferingChangeRequestId application offering change request id.
    * @param payload information to update the application offering change request.
+   * @param userToken user approving or declining the application offering change request.
    */
+  @Roles(Role.InstitutionApproveDeclineApplicationOfferingChangeRequest)
   @Patch(":applicationOfferingChangeRequestId")
+  @ApiNotFoundResponse({
+    description:
+      "Application offering change not found or not in valid status to be updated.",
+  })
   async updateApplicationOfferingChangeRequest(
     @Param("applicationOfferingChangeRequestId", ParseIntPipe)
     applicationOfferingChangeRequestId: number,
     @Body()
     payload: ApplicationOfferingChangeAssessmentAPIInDTO,
+    @UserToken() userToken: IUserToken,
   ): Promise<void> {
-    // TODO: API including the addition of ministry role to be done in the next PR.
+    const applicationOfferingChangeRequest =
+      await this.applicationOfferingChangeRequestService.applicationOfferingChangeRequestExists(
+        applicationOfferingChangeRequestId,
+        {
+          applicationOfferingChangeRequestStatus:
+            ApplicationOfferingChangeRequestStatus.InProgressWithSABC,
+        },
+      );
+    if (!applicationOfferingChangeRequest) {
+      throw new NotFoundException(
+        "Application offering change not found or not in valid status to be updated.",
```

Comment:
to be approved/declined ?

Same error message is found in students so it will be good to be specific here.